### ISSUE TITLE
NMO respawns only one player when zombies near

### ISSUE THREAD

**Issue by Lunreth** [nmrih/source-game#1131]
This issue may be linked with the other one we closed, but i don't find any relation between the two so here it goes:

Exactly what title says, when playing NMO maps you reach a respawn point but then only one player will respawn (even when maps use 8 respawn entities) because zombies are too near. However, the other players will have to wait until the area is cleared for them to respawn one by one (it will appear "waiting for spawn point"), this doesn't seem fair since 1 random player will get the chance to play while the others watch. They will eventually respawn, so it doesn't seem to be critical. but still, seems like an issue for some of them and they wanted me to publish this.

Steps to reproduce bug:
1 - Set cvars (which seem to be default according to previous closed issue)
sv_respawn_allowed 0
sv_respawn_nearby_allowed 0
sv_respawn_nearby_npc_radius 192
2 - Play multiplayer "nmo_medical_clinic" or any other map with huge density of zombies near player respawn point
3 - Reach respawn point with 2 or more death players to see how only 1 respawns (keep in mind that this map has 8 respawn points, checked via decompiling)

![medical1](https://user-images.githubusercontent.com/26940632/76693128-1b95db00-663f-11ea-86e7-408ce7a8c965.jpg)

NOTE: this issue only happens in certain maps like the one above, that is why i think its the zombie density, but could be something else, maybe i'm wrong.

**Comment by dysphie** [comment]
Suggestion: remove this nearby check entirely, it's never done anything but cause issues for mappers and players.

**Comment by felis-catus** [comment]
> Suggestion: undo this feature entirely #590, it's never done anything but cause issues for mappers and players.

Agreed to disable the primary check (npc limit) and keep the secondary (radius).

**Comment by felis-catus** [comment]
Resolved in src-r4933.

**Comment by felis-catus** [comment]
Re-opening because the issue is still prevalent. (Maybe remove the secondary radius check as well?)

**Comment by overmase** [comment]
Enabling sv_respawn_nearby_allowed still causes problems

sv_respawn_allowed is enabled, spawn conditions may be met, but "Wait for spawn ..." may cause a stuck
When this bug occurs, players will not be able to spawn even though there are checkpoints.

Easy way to reproduce

Case 1

1: Play nmo_junction and head to the sewer with two or more players

2: One player commits suicide there and the other waits there to spawn

keep spawning to nearby players despite having checkpoints, but I can't spawn 
However can spawn when the player returns to the flat area
I think this is due to the unstable terrain

This looks like a bug because nmo_suzhou will spawn at checkpoints if nearby players can't spawn.


Case 2

enable spawn nearby and info_player_nmrih is blocked by zombies, you will not be able to spawn regardless of distance, even if it is safe near the player.

1: Play nmo_cabin　or nmo_underground to block the first info_player_nmrih with zombies

2: Wait in a place where one player can safely spawn, another player commits suicide and waits for the spawn

Case 2 may not be able to spawn permanently
![iss3](https://user-images.githubusercontent.com/20251057/160260972-54c8bf32-d0ea-4964-91a1-36f3bc54b56f.png)
